jax/_src/test_util.py

```diff
@@ -468,6 +468,65 @@ def is_device_tpu(version: int | None = None, variant: str = "") -> bool:
     return "v6 lite" in device_kind
   return expected_version in device_kind
 
+def pattern_search(patterns: str | Sequence[str], string: str):
+  if isinstance(patterns, list):
+    patterns = tuple(patterns)
+  elif not isinstance(patterns, tuple):
+    patterns = (patterns,)  # type: ignore
+
+  for pattern in patterns:
+    if re.search(pattern, string):
+      return pattern
+  return None
+
+def skip_if_device_kind_matches(device_patterns: str | Sequence[str]):
+  """A decorator for test methods to skip the test when run on specified devices."""
+  device_kind = jax.devices()[0].device_kind
```

Comment:
This is breaking our internal CI - can you move this line and the one below inside of the `skip` function below? We can't query jax.devices() at the top-level of a python file.